src/configcontainer.cpp

```diff
@@ -20,629 +19,139 @@ namespace newsboat {
-	{"miniflux-min-items", ConfigData("100", ConfigDataType::INT)},
-	{"miniflux-password", ConfigData("", ConfigDataType::STR)},
-	{"miniflux-passwordfile", ConfigData("", ConfigDataType::PATH)},
-	{"miniflux-passwordeval", ConfigData("", ConfigDataType::STR)},
-	{
-		"miniflux-show-special-feeds",
-		ConfigData("yes", ConfigDataType::BOOL)},
-	{"miniflux-token", ConfigData("", ConfigDataType::STR)},
-	{"miniflux-tokenfile", ConfigData("", ConfigDataType::PATH)},
-	{"miniflux-tokeneval", ConfigData("", ConfigDataType::STR)},
-	{"miniflux-url", ConfigData("", ConfigDataType::STR)},
-	{
-		"urls-source",
-		ConfigData("local",
-		std::unordered_set<std::string>({"local",
-			"opml",
-			"oldreader",
-			"ttrss",
-			"newsblur",
-			"feedhq",
-			"feedbin",
-			"freshrss",
-			"ocnews",
-			"miniflux",
-			"inoreader"}))},
-	{"use-proxy", ConfigData("no", ConfigDataType::BOOL)},
-	{"user-agent", ConfigData("", ConfigDataType::STR)},
-
-	/* title formats: */
-	{
-		"articlelist-title-format",
-		ConfigData(_("%N %V - Articles in feed '%T' (%u unread, %t total)%?F? matching filter '%F'&? - %U"),
-			ConfigDataType::STR)},
-	{
-		"dialogs-title-format",
-		ConfigData(_("%N %V - Dialogs"), ConfigDataType::STR)},
-	{
-		"feedlist-title-format",
-		ConfigData(_("%N %V - %?F?Feeds&Your feeds? (%u unread, %t total)%?F? matching filter '%F'&?%?T? - tag '%T'&?"),
-			ConfigDataType::STR)},
-	{
-		"filebrowser-title-format",
-		ConfigData(_("%N %V - %?O?Open File&Save File? - %f"),
-			ConfigDataType::STR)},
-	{
-		"dirbrowser-title-format",
-		ConfigData(_("%N %V - %?O?Open Directory&Save File? - %f"),
-			ConfigDataType::STR)},
-	{
-		"help-title-format",
-		ConfigData(_("%N %V - Help"), ConfigDataType::STR)},
-	{
-		"itemview-title-format",
-		ConfigData(_("%N %V - Article '%T' (%u unread, %t total)"),
-			ConfigDataType::STR)},
-	{
-		"searchresult-title-format",
-		ConfigData(_("%N %V - Search results for '%s' (%u unread, %t total)%?F? matching filter '%F'&?"),
-			ConfigDataType::STR)},
-	{
-		"selectfilter-title-format",
-		ConfigData(_("%N %V - Select Filter"),
-			ConfigDataType::STR)},
-	{
-		"selecttag-title-format",
-		ConfigData(_("%N %V - Select Tag"),
-			ConfigDataType::STR)},
-	{
-		"urlview-title-format",
-		ConfigData(_("%N %V - URLs"), ConfigDataType::STR)},
-	{
-		"wrap-scroll",
-		ConfigData("no", ConfigDataType::BOOL)},
-}
+	: rs_object(newsboat::configcontainer::bridged::create())
 {
 }
 
 ConfigContainer::~ConfigContainer() = default;
 
 void ConfigContainer::register_commands(ConfigParser& cfgparser)
 {
-	// this registers the config options defined above in the configuration
-	// parser
-	// -> if the resp. config option is encountered, it is passed to the
-	// ConfigContainer
-	std::lock_guard<std::recursive_mutex> guard(config_data_mtx);
-	for (const auto& cfg : config_data) {
-		cfgparser.register_handler(cfg.first, *this);
+	const auto keys = newsboat::configcontainer::bridged::get_suggestions(*rs_object, "");
```

Comment:
Nit: it'd be nice to have a comment here explaining that `get_suggestions("")` returns names of all available options, so it's clear what's going on. Reading this code for the first time I was surprised that suggestions have anything to do with command registrations.

Also, I like how this cleverly sidesteps the need to call C++ from Rust :)